lib/wasix/src/runners/dproxy/hyper_proxy/stream.rs

```diff
@@ -0,0 +1,93 @@
+use std::io;
+
+use futures::Stream;
+use hyper::client::connect::Connected;
+use tokio_stream::wrappers::BroadcastStream;
+use virtual_net::tcp_pair::{TcpSocketHalfRx, TcpSocketHalfTx};
+
+use super::*;
+
+#[derive(Debug)]
+pub struct HyperProxyStream {
+    pub(super) tx: TcpSocketHalfTx,
```

Comment:
I'm not entirely sure why we need a special DProxy runner rather than just a journaled runner.

Is it primarily for the sharding?
And/or snapshotting after each request?

Reply:
The dproxy runner spawns and reuses instances while the dcgi runner creates them and throws them away. they are very different operating models. Dproxy actually proxies the requests as well while dcgi does not